generated/nidcpower/session.py

```
@@ -4151,6 +4151,599 @@ def export_attribute_configuration_file(self, file_path):
+                'property': self.current_limit_high,
+            },
+            "current_limit_low": {
+                'id': 1150188,
+                'property': self.current_limit_low,
+            },
+            "pulse_voltage_limit_high": {
+                'id': 1150189,
+                'property': self.pulse_voltage_limit_high,
+            },
+            "pulse_voltage_limit_low": {
+                'id': 1150190,
+                'property': self.pulse_voltage_limit_low,
+            },
+            "pulse_bias_voltage_limit_high": {
+                'id': 1150191,
+                'property': self.pulse_bias_voltage_limit_high,
+            },
+            "pulse_bias_voltage_limit_low": {
+                'id': 1150192,
+                'property': self.pulse_bias_voltage_limit_low,
+            },
+            "pulse_current_limit_high": {
+                'id': 1150193,
+                'property': self.pulse_current_limit_high,
+            },
+            "pulse_current_limit_low": {
+                'id': 1150194,
+                'property': self.pulse_current_limit_low,
+            },
+            "pulse_bias_current_limit_high": {
+                'id': 1150195,
+                'property': self.pulse_bias_current_limit_high,
+            },
+            "pulse_bias_current_limit_low": {
+                'id': 1150196,
+                'property': self.pulse_bias_current_limit_low,
+            },
+            "voltage_level": {
+                'id': 1250001,
+                'property': self.voltage_level,
+            },
+            "ovp_enabled": {
+                'id': 1250002,
+                'property': self.ovp_enabled,
+            },
+            "ovp_limit": {
+                'id': 1250003,
+                'property': self.ovp_limit,
+            },
+            "current_limit": {
+                'id': 1250005,
+                'property': self.current_limit,
+            },
+            "output_enabled": {
```

I think it's unfortunate that we are duplicating all this information we already have within the Session class just in dictionary form. I understand that since we use a code-generator, it doesn't add any maintenance burden. But it just seems _wrong_. We should be looking up the member of the class Session by the given name, then trying to get the attribute ID. If that fails, we can raise the appropriate exception.

---

I am trying to keep everything isolated to this method. To do what you suggest would require modifying the attributes to allow getting the attribute ID without sneaking in and reading a 'private' variable.

---

>  To do what you suggest would require modifying the attributes to allow getting the attribute ID without sneaking in and reading a 'private' variable.

I think that's acceptable. It's all within the same module.

---

Same module but different classes.

I will see what I can do so we can see what that would look like.